Sources/Classes/RSDeviceInfo.m

```diff
@@ -33,6 +33,10 @@ - (instancetype) initWithConfig: (RSConfig *) config {
         _model = [self getDeviceModel];
         _manufacturer = @"Apple";
         _attTrackingStatus = RSATTNotDetermined;
+        _advertisingId = [[RSPreferenceManager getInstance] getAdvertisingId];
+        if(_advertisingId != nil) {
+            _adTrackingEnabled = YES;
+        }
```

Comment:
I would suggest to ensure that `advertisingId` is valid and then only set this `_adTrackingEnabled = YES`.

```suggestion
        if(_advertisingId != nil && ![_advertisingId isEqualToString:@"00000000-0000-0000-0000-000000000000"]) {
            _adTrackingEnabled = YES;
        }
```

Reply:
Updated the logic to push advertisingId to push to defaults only if it is valid, so we can skip this now.